src/Stryker.Core/Stryker.Core.UnitTest/Reporters/Html/Realtime/SseServerTest.cs

```diff
@@ -1,21 +1,54 @@
 using System;
+using System.Diagnostics;
 using System.Threading;
 using System.Threading.Tasks;
 using LaunchDarkly.EventSource;
+using Shouldly;
 using Stryker.Core.Reporters.Html.Realtime;
 using Stryker.Core.Reporters.Html.Realtime.Events;
 using Xunit;
+using Xunit.Sdk;
 
 namespace Stryker.Core.UnitTest.Reporters.Html.Realtime;
 
 public class SseServerTest : TestBase
 {
-    private readonly ISseServer _sut;
+    private readonly SseServer _sut;
+    private readonly object _lock = new();
+    private bool _connected;
 
-    public SseServerTest() => _sut = new SseServer();
+    public SseServerTest()
+    {
+        _sut = new SseServer();
+        _sut.ClientConnected += ClientConnected;
+    }
+
+    private void ClientConnected(object sender, EventArgs e)
+    {
+        lock (_lock)
+        {
+            _connected = true;
+            Monitor.Pulse(_lock);
+        }
+    }
+
+    private bool WaitConnection(int timeout)
+    {
+        var watch = new Stopwatch();
+        watch.Start();
+        lock (_lock)
+        {
+            while (!_connected && watch.ElapsedMilliseconds < timeout)
+            {
+                Monitor.Wait(_lock, Math.Max(timeout - (int)watch.ElapsedMilliseconds, 1));
+            }
```

Comment:
Is the while loop necessary? What if we only do this:

```suggestion
            Monitor.Wait(_lock, Math.Max(timeout, 1));
```

Should be functionally equivalent, no?